### ISSUE TITLE
Batch API should be able to acquire failures every action

### ISSUE THREAD

**Issue by tkrs** [tkrs/orcus#122]
I want to return the `table.batch` to `Iterator[F[Either[Throwable, T]]]`.

**Comment by tkrs** [comment]
Similarly, I want to `table.batch`'s return type to `F[Vector[Either[Throwable, ?]]]`.

**Comment by tkrs** [comment]
Umm, I guess `Either[Throwable, Option[Result]]` is a redendet...

**Comment by tkrs** [comment]
I think that it is better to make an ADT for the batch result.

**Comment by tkrs** [comment]
```scala
sealed trait Ret
case class Err(t: Throwable) extends Ret
case class HRet(r: Result) extends Ret
case object Void extends Ret
```

**Comment by tkrs** [comment]
Fixed by #123
